langchain-core/src/prompts/image.ts

```diff
@@ -0,0 +1,186 @@
+      // checkValidTemplate(
+      //   this.template,
+      //   this.templateFormat,
+      //   totalInputVariables
+      // );
+    }
+  }
+
+  _getPromptType(): "prompt" {
+    return "prompt";
+  }
+
+  /**
+   * Partially applies values to the prompt template.
+   * @param values The values to be partially applied to the prompt template.
+   * @returns A new instance of ImagePromptTemplate with the partially applied values.
+   */
+  async partial<NewPartialVariableName extends string>(
+    values: PartialValues<NewPartialVariableName>
+  ) {
+    const newInputVariables = this.inputVariables.filter(
+      (iv) => !(iv in values)
+    ) as Exclude<Extract<keyof RunInput, string>, NewPartialVariableName>[];
+    const newPartialVariables = {
+      ...(this.partialVariables ?? {}),
+      ...values,
+    } as PartialValues<PartialVariableName | NewPartialVariableName>;
+    const promptDict = {
+      ...this,
+      inputVariables: newInputVariables,
+      partialVariables: newPartialVariables,
+    };
+    return new ImagePromptTemplate<
+      InputValues<
+        Exclude<Extract<keyof RunInput, string>, NewPartialVariableName>
+      >
+    >(promptDict);
+  }
+
+  /**
+   * Formats the prompt template with the provided values.
+   * @param values The values to be used to format the prompt template.
+   * @returns A promise that resolves to a string which is the formatted prompt.
+   */
+  async format<FormatOutput = ImageURL>(
+    values: TypedPromptInputValues<RunInput>
+  ): Promise<FormatOutput> {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const formatted: Record<string, any> = {};
+    for (const [key, value] of Object.entries(this.template)) {
+      if (typeof value === "string") {
+        formatted[key] = value.replace(/{([^{}]*)}/g, (match, group) => {
+          const replacement = values[group];
+          return typeof replacement === "string" ||
+            typeof replacement === "number"
+            ? String(replacement)
+            : match;
+        });
+      } else {
+        formatted[key] = value;
+      }
+    }
+    let url = values.url || formatted.url;
+    const path = values.path || formatted.path;
+    const detail = values.detail || formatted.detail;
+    if (!url && !path) {
+      throw new Error("Must provide either url or path.");
+    }
+    if (!url) {
+      if (typeof path !== "string") {
+        throw new Error("path must be a string.");
+      }
+      // Requires node:fs so we don't want to always
+      // import for client side application support.
+      const { imageToDataUrl } = await ImagePromptTemplate.imports();
+      url = await imageToDataUrl(path);
+    }
+    if (typeof url !== "string") {
+      throw new Error("url must be a string.");
+    }
+    const output: ImageURL = { url };
+    if (detail) {
+      output.detail = detail;
+    }
+    return output as FormatOutput;
+  }
+
+  /** @ignore */
+  static async imports(): Promise<{
```

Comment:
We don't do this anymore?

Reply:
What's the alternative to conditionally import?